File: famousfront/datamodels/FeedEntry.cs
using System.Runtime.Serialization;

namespace famousfront.datamodels
{
  [DataContract]
  internal class FeedEntry
  {
    [DataMember(Name = "_id")]
    public string id
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public uint flags
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public string source
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public uint type
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public string uri
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public FeedTitle title
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public FeedAuthor author
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public long pubdate
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public string summary
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public string content
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public string[]tags
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public FeedMedia[] images
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public FeedMedia[]videos
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public FeedMedia[]audios
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public FeedLink[] links
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public uint words
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public uint density
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public ulong status
    {
      get;set;
    }
    [DataMember(EmitDefaultValue = false)]
    public string[] categories
    {
      get;set;
    }
  }
}

File: src/github.com/heartszhang/feed/feedentry.go
package feed

type FeedEntry struct {
	Parent     string      `json:"src,omitempty" bson:"src,omitempty"` // feed source's uri
	Uri        string      `json:"uri,omitempty" bson:"uri,omitempty"`
	Summary    string      `json:"summary,omitempty" bson:"summary,omitempty"`
	Content    string      `json:"content,omitempty" bson:"content,omitempty"`
	Type       uint        `json:"type" bson:"type"` // feed_type...
	Title      string      `json:"title,omitempty" bson:"title,omitempty"`
	Author     string      `json:"author,omitempty" bson:"author,omitempty"`
	PubDate    int64       `json:"pubdate" bson:"pubdate"` // unix time
	Tags       []string    `json:"tags,omitempty" bson:"tags,omitempty"`
	Categories []string    `json:"categories,omitempty" bson:"category,omitempty"`
	Images     []FeedMedia `json:"images,omitempty" bson:"images,omitempty"`
	Videos     []FeedMedia `json:"videos,omitempty" bson:"videos,omitempty"`
	Audios     []FeedMedia `json:"audios,omitempty" bson:"audios,omitempty"`
	//	Links      []FeedLink    `json:"links,omitempty" bson:"links,omitempty"`
}

type FeedEntity struct {
	FeedSource `json:",inline"`
	Entries    []FeedEntry `json:"entries,omitempty"`
}
